Add requests as a legit dependency

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25596 Add requests as a legit dependency**

Giving up on trying to limit this to just a py2 dependency

Differential Revision: [D17171063](https://our.internmc.facebook.com/intern/diff/D17171063)

Commit:
Add requests as a legit dependency

pietern commented on requirements.txt:
```diff
@@ -4,3 +4,4 @@ pyyaml
 setuptools
 six
 typing
+requests
```
Seems to be ordered alphabetically.

pietern reviewed (commented):
LGTM, but will defer approval to @ezyang or @soumith since this adds a new dep.

Commit:
Update on "Add requests as a legit dependency"


Giving up on trying to limit this to just a py2 dependency

Commit:
Update on "Add requests as a legit dependency"


Giving up on trying to limit this to just a py2 dependency

Commit:
Update on "Add requests as a legit dependency"


Giving up on trying to limit this to just a py2 dependency

soumith commented:
but why?

jamesr66a commented:
@soumith because this seems like a pretty poor experience: https://github.com/pytorch/pytorch/pull/25083#issuecomment-527219928

Commit:
Update on "Add requests as a legit dependency"


Giving up on trying to limit this to just a py2 dependency

Differential Revision: [D17171063](https://our.internmc.facebook.com/intern/diff/D17171063)

xuhdev commented on torch/hub.py:
```diff
@@ -374,24 +368,11 @@ def load(github, model, *args, **kwargs):
-        if content_length is not None and len(content_length) > 0:
-            file_size = int(content_length[0])
+    content_length = response.headers['Content-Length']
+    file_size = content_length
```
Actually I don't see how `content_length` is used. Perhaps we should remove it?

soumith reviewed (changes requested):
can you please add requests to requirements.txt and to README but leave the code unchanged.
Once we switch to Python3 when Py2 EOLs, we can remove this dependency with the current code.

Commit:
Update on "Add requests as a legit dependency"


Giving up on trying to limit this to just a py2 dependency

Differential Revision: [D17171063](https://our.internmc.facebook.com/intern/diff/D17171063)

jamesr66a commented:
@soumith does this look OK now?

soumith reviewed (approved):
thanks lgtm
